tests/unit/test_rehearse_first_order_script.py

```
@@ -0,0 +1,140 @@
+    assert "FAIL" in stdout
+    assert "audit events were" in stdout
+
+
+def test_rehearsal_result_dataclass_is_frozen() -> None:
+    """STO-10: result-bearing value objects are frozen dataclasses
+    (project convention; see CLAUDE.md). Prevents accidental post-hoc
+    mutation of an audit-bearing record."""
+    result = RehearsalResult(
+        passed=True,
+        events=["approval_matched"],
+        audit_path=Path("/tmp/audit.jsonl"),
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/audit.jsonl"

(S108)